### PR TITLE
fix: show token icon in transaction list and show balance in token list

### LINKED ISSUE #3572: [Bug] Use debank openAPI for token icon instead of Trust wallet icon
## Bug Info

Screen Shot 2021-07-05 at 03 03 33
https://openapi.debank.com/docs

Trust wallet icon url is outdated. use debank open api since we're using it already 
Screen Shot 2021-07-05 at 03 14 04

### How to reproduce?

1.
2.
3.

### Mask Version:

- [ ] (Browser) Installed from the extension store (version: )
- [ ] (App) Installed from the app store (version: )
- [ ] master branch
- [ ] Build of version/commit hash:

(Please write down build command/flags if there is anything special)

### System

 
- [ ] Windows 10
- [ ] Mac OS X Catalina
- [ ] Linux: 

### Platform/Browser

 
- [ ] Latest Chrome (stable channel)
- [ ] Latest Firefox (stable channel)
- [ ] Android 11
- [ ] iOS 14

### PR DESCRIPTION
**Pull request by NikoManiac**
closes #3572

### CODE REVIEW

File: packages/maskbook/src/extension/background-script/EthereumServices/tokenList.ts [hunk 1]
```diff
@@ -6,75 +6,123 @@ import {
...
+async function fetch1inchERC20TokensFromTokenList(
     url: string,
     chainId = ChainId.Mainnet,
 ): Promise<ERC20TokenDetailed[]> {
-    return (await fetchTokenList(url)).tokens
-        .filter((x) => x.chainId === chainId && isChainIdValid(chainId, Flags.wallet_allow_testnet))
+    const tokens = ((await fetchTokenList(url)) as TokenObject).tokens
+    const _tokens = Object.values(tokens)
+    return _tokens
+        .filter((x) => x.address.toLowerCase() !== NATIVE_TOKEN_ADDRESS_IN_1INCH)
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/background-script/EthereumServices/tokenList.ts:62]: You can filter the ether token out in the `fetchCommonERC20TokensFromTokenList`. There is no reason an ERC20 token will use the address `0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee`.

File: packages/maskbook/src/extension/options-page/DashboardComponents/FixedTokenList.tsx [hunk 2]
```diff
@@ -42,18 +47,43 @@ export function FixedTokenList(props: FixedTokenListProps) {
...
+    const {
+        loading: loadingAssets,
+        value: assets,
+        error,
+    } = useAssetsFromChain(renderTokens.filter((x) => EthereumAddress.isValid(x.address)))
+
+    const renderAssets =
+        error || !account || loadingAssets
+            ? renderTokens.map((token) => ({ token: token, balance: null }))
+            : assets.sort((a, b) => {
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/options-page/DashboardComponents/FixedTokenList.tsx:82]: Actually, we've got a bug here. The demand is to render the MASK token in the first place. But we sorted twice here. The second one will override the previous result. Could you merge them into one sorting?

File: packages/maskbook/src/extension/options-page/DashboardComponents/TokenInList.tsx [hunk 3]
```diff
@@ -75,7 +72,11 @@ export function TokenInList({ data, index, style }: TokenInListProps) {
 
     const stop = useCallback((ev: React.MouseEvent<HTMLAnchorElement>) => ev.stopPropagation(), [])
 
-    const token = data.tokens[index]
+    const currentAsset = data.assets[index]
+    const token = currentAsset.token
+    const balance = currentAsset.balance
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/options-page/DashboardComponents/TokenInList.tsx:77]:
```suggestion
    const { token, balance } = currentAsset
```

File: packages/maskbook/src/extension/options-page/DashboardComponents/TokenInList.tsx [hunk 4]
```diff
@@ -89,12 +90,6 @@ export function TokenInList({ data, index, style }: TokenInListProps) {
...
             <ListItemText classes={{ primary: classes.text }}>
                 <Typography className={classes.primary} color="textPrimary" component="span">
-                    <span className={classes.name}>{name}</span>
-                    <span className={classes.address}>
-                        {token.address !== NATIVE_TOKEN_ADDRESS ? formatEthereumAddress(token.address, 8) : null}
-                    </span>
-                </Typography>
-                <Typography className={classes.secondary} color="textSecondary" component="span">
                     <span className={classes.symbol}>{symbol}</span>
                     {token.address !== NATIVE_TOKEN_ADDRESS ? (
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/options-page/DashboardComponents/TokenInList.tsx:94]: Use `isSameAddress` instead.

File: packages/maskbook/src/extension/background-script/EthereumServices/tokenList.ts (hunk 1 above)
Comment (NikoManiac) [reply packages/maskbook/src/extension/background-script/EthereumServices/tokenList.ts:62]: The 1inch token have different structure: 
```javascript
'adddress': {
  address: '',
  logoURI: '',
  ...ohter,
}
```
So, I need new method to handle this.

**Comment by guanbinrui** [comment]
@NikoManiac Should rebase this PR again to remove unrelated commits.

File: packages/maskbook/src/extension/options-page/DashboardComponents/FixedTokenList.tsx (hunk 2 above)
Comment (NikoManiac) [reply packages/maskbook/src/extension/options-page/DashboardComponents/FixedTokenList.tsx:82]: OK

File: packages/maskbook/src/extension/background-script/EthereumServices/tokenList.ts (hunk 1 above)
Comment (guanbinrui) [reply packages/maskbook/src/extension/background-script/EthereumServices/tokenList.ts:62]: It tells that the 1inch token list did not match the spec. 🤔

Comment (guanbinrui) [reply packages/maskbook/src/extension/background-script/EthereumServices/tokenList.ts:62]: I see. It was using the 1inch token list API instead.

File: packages/maskbook/src/extension/options-page/DashboardComponents/WalletAssetsTable.tsx [hunk 5]
```diff
@@ -187,6 +188,7 @@ export interface WalletAssetsTableProps extends withClasses<never> {
 export function WalletAssetsTable(props: WalletAssetsTableProps) {
     const { t } = useI18N()
     const { wallet } = props
+    console.log(wallet)
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/options-page/DashboardComponents/WalletAssetsTable.tsx:191]:
```suggestion
```

File: packages/maskbook/src/web3/UI/EthereumChainBoundary.tsx [hunk 6]
```diff
@@ -37,6 +37,8 @@ export function EthereumChainBoundary(props: EthereumChainBoundaryProps) {
 
     // if false then it will not guide the user to switch the network
     const isAllowed = chainIdValid && !!account
+    // if false then the user should switch network manually
+    const isSwitchable = true
```
Comment (guanbinrui) [inline packages/maskbook/src/web3/UI/EthereumChainBoundary.tsx:41]:
```suggestion
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**